modules/generator/storage/instance.go

```diff
@@ -57,12 +57,19 @@ func New(cfg *Config, o Overrides, tenant string, reg prometheus.Registerer, log
 
 	walDir := filepath.Join(cfg.Path, tenant)
 
+	// clean the wal before everything
+	level.Info(logger).Log("msg", "clearing old WAL on start up", "dir", walDir)
```

Comment:
any reason not to just do this once in generator.go instead of per tenant?

Reply:
when I moved it to the very first line of code that touches the generator, the test kept failing because in the config file, we set the wal path to `/var/tempo` which is the same path for the other components -- and this would remove other folders created by other components. We normally set our walp ath to `/var/tempo/wal` but I was thinking... what if there are users out there that store their wal folders in the same path as their other components. Clearing the entire folder will nuke everything else they might have stored there so it will be safer to remove exactly only what we want to remove.

Reply:
+1 good call